tt_metal/hw/ckernels/wormhole_b0/metal/llk_io/llk_io_unpack.h

```diff
@@ -52,6 +73,8 @@ inline void llk_wait_tiles(int operand, std::int32_t num_tiles) {
         tiles_received = (std::uint16_t) reg_read((std::uint32_t)tiles_received_ptr);
         num_tiles_recv = tiles_received - cb_interface[input].tiles_acked;
     } while (num_tiles_recv < num_tiles_u);
+
+    apply_mm_stagger(operand);
```

Comment:
If somehow this PR gets approved, maybe put this into matmul kernels that call this as opposed to putting it in a general llk api. Maybe that will fix your code compile issues.

Reply:
@aliuTT  there is no need for this type of language.
You are required reviewer for this PR you can prevent this from going in.
If you have arguments why this shouldn't go in present them here, we can have a larger discussion about this.
This is in no way clean solution that fully resolves the problem. This is just best effort to provide model developers with as much stability as we can, and provide with tools to trade stability/perf.

Regarding your technical comment here.
Initial ask was to not touch kernels and to "hide" this into LLKs but I would agree with you that it would be better to move this out to mm compute kernel.

Reply:
The stagger code should be exclusively hidden inside llk api and not exposed to developers.
the application of the stagger is something that developers should be able to control, based on the guidance provided by LLK team.

Also, the goal is to make this non-intrusive for kernels that don't need to apply the stagger, therefore have the function be compile-time optimized in cases when we dont need it.
The cleanliness of the solution is definitelly something we can work on based on constructive feedback, but you have to keep in mind that this is a pretty heavy workaround and its hard to hide it entirely. I anticipate this to be an iterative solution.